src/java.base/share/classes/sun/security/provider/SunEntries.java

```diff
@@ -100,10 +100,40 @@
 public final class SunEntries {
 
     /* The property 'jdk.nativeDigest' is used to control enablement of the native
-     * digest implementation.
+     * digest implementations.
      */
     private static final boolean useNativeDigest = NativeCrypto.isAlgorithmEnabled("jdk.nativeDigest", "MessageDigest");
 
+    /* The property 'jdk.nativeMD5' is used to control enablement of the native
+     * MD5 implementations.
+     */
+    private static final boolean useNativeMD5 = NativeCrypto.isAlgorithmEnabled("jdk.nativeMD5", "MD5");
```

Comment:
Suggest each of these new flags should use `useNativeDigest` in their initialization. For example:
```java
    private static final boolean useNativeMD5 = useNativeDigest && NativeCrypto.isAlgorithmEnabled("jdk.nativeMD5", "MD5");
```
Later the code need only ask `useNativeMD5` instead of `useNativeDigest && useNativeMD5`. It also avoids most calls to `NativeCrypto.isAlgorithmEnabled()` if `useNativeDigest` is false.

The initializations could be captured in a single `static` block where `useNativeDigest` is only a local variable (there would be no references to `useNativeDigest` afterward).

Reply:
Changed to suggestion.